### ISSUE TITLE
How to use this with pytest-django?

### ISSUE THREAD

**Issue by auvipy** [pytest-dev/pytest-bdd#161]

**Comment by olegpidsadnyi** [comment]
The same way as you use pytest-django. Pytest-bdd will only allow you to compose your "requirements" in form of gherkin feature files. Basically use bdd scenarios instead of the django TestCases.

I'm not using pytest-django because I'm using SQLAlchemy ORM and therefore there are custom helpers and fixtures that configure the database, wsgi applications etc.
More than that I'm using pytest-splinter - the real browser interaction for the high-level functional tests.
If your interfaces don't have a lot of javascript or you are working on the API project then pytest-django RequestFactory is probably enough. It is just a question if you want to implement BDD steps and reuse them between your tests or do you want to do the composition of the code on Python level.

**Comment by auvipy** [comment]
Well I use django with it's ORM and django rest framework for API. I planned to used pytest-bdd and pytest-splinter, for client side I planned to use vue.js+coffeescript, webpack and karma runner will be used. any other suggested tools you do feel I'm missing?

**Comment by olegpidsadnyi** [comment]
I see. It is possible as well. You would have to use http://pytest-django.readthedocs.org/en/latest/helpers.html#live-server live_server fixture. Splinter is a real browser so it needs to open URL. You can get the base URL from live_server.url

**Comment by bubenkoff** [comment]
@auvipy here's an example you can use
https://github.com/paylogic/pdt
i used pytest-bdd for this django project

**Comment by paultiplady** [comment]
pytest-bdd doesn't work out of the box with `pytest-django`. This issue is currently the top google result when searching for solutions, so I'm linking here to save future travelers from straying from the correct path.

This issue provides a workaround for the problem: https://github.com/pytest-dev/pytest-bdd/issues/215

You can either use `transactional_db` explicitly, or jump through some hoops to use it automatically when generating tests.

There's another bug report here complaining about what looks like the same issue that I hit. https://github.com/pytest-dev/pytest-bdd/issues/187

**Comment by bubenkoff** [comment]
@auvipy @paultiplady please check how I used it, worked quite ok: https://github.com/paylogic/pdt
might be good to add as a reference to the docs

**Comment by auvipy** [comment]
cool thanks

**Comment by auvipy** [comment]
@olegpidsadnyi I would request to re-open this issue as docs improvement, I would like to contribute detail docs from the real project of @bubenkoff 's https://github.com/paylogic/pdt

**Comment by paultiplady** [comment]
FWIW I found it a bit hard to follow the https://github.com/paylogic/pdt example from the codebase; it all looked simple enough, but I didn't see how it resolved the problem I was hitting. It would be very useful to distill it down to the minimum viable to actually get `pytest-django` working.

**Comment by foobacca** [comment]
https://github.com/paylogic/pdt is giving a 404 for me. Is there a new URL? Or another project that could serve as an example?
